src/olympia/users/serializers.py

```diff
@@ -12,7 +14,20 @@ class Meta:
         fields = ('id', 'name', 'url')
 
     def get_url(self, obj):
-        return absolutify(obj.get_url_path())
+        def is_developer():
+            return (self.context['is_developer']
+                    if 'is_developer' in self.context
+                    else obj.is_developer)
+
+        def is_adminish(user):
+            return user and acl.action_allowed_user(user,
+                                                    amo.permissions.USERS_EDIT)
+
+        # Only return your own profile url, and for developers.
+        current_user = getattr(
+            self.context.get('request', object()), 'user', None)
+        if obj == current_user or is_adminish(current_user) or is_developer():
+            return absolutify(obj.get_url_path())
```

Comment:
I'm not sure your `is_adminish` replacement is better... you're returning user rather than a bool and skipping the check that user isn't None (the `user and...` part)